gluoncv/utils/lr_scheduler.py

```diff
@@ -5,104 +5,114 @@
+    def add(self, scheduler):
+        assert(isinstance(scheduler, LRScheduler))
 
-    For warmup_mode='linear'::
+        scheduler.offset = self.count
+        self.count += scheduler.niters
+        self.update_sep.append(self.count)
+        self.schedulers.append(scheduler)
 
-        lr = warmup_lr + (baselr - warmup_lr) * iter / max_warmup_iter
+    def __call__(self, num_update):
+        self.update(num_update)
+        return self.learning_rate
 
-    For warmup_mode='constant'::
+    def update(self, num_update):
+        if num_update < self.count:
+            ind = len(self.schedulers) - 1
+            for i, sep in enumerate(self.update_sep):
+                if sep > num_update:
+                    ind = i - 1
+                    break
+            lr = self.schedulers[ind]
+            lr.update(num_update)
+            self.learning_rate = lr.learning_rate
 
-        lr = warmup_lr
+class LRScheduler(lr_scheduler.LRScheduler):
+    r"""Learning Rate Scheduler
 
     Parameters
     ----------
+
     mode : str
         Modes for learning rate scheduler.
-        Currently it supports 'step', 'poly' and 'cosine'.
+        Currently it supports 'constant', 'step', 'linear', 'poly' and 'cosine'.
     baselr : float
         Base learning rate, i.e. the starting learning rate.
+    targetlr : float
+        Target learning rate, i.e. the ending learning rate.
+        With constant mode targetlr is ignored.
     niters : int
-        Number of iterations in each epoch.
-    nepochs : int
-        Number of training epochs.
+        Number of iterations to be scheduled.
+    offset : int
+        Number of iterations before this scheduler.
+    power : float
+        Power parameter of poly scheduler.
     step : list
-        A list of epochs to decay the learning rate.
+        A list of iterations to decay the learning rate.
     step_factor : float
         Learning rate decay factor.
-    targetlr : float
-        Target learning rate for poly and cosine, as the ending learning rate.
-    power : float
-        Power of poly function.
-    warmup_epochs : int
-        Number of epochs for the warmup stage.
-    warmup_lr : float
-        The base learning rate for the warmup stage.
-    warmup_mode : str
-        Modes for the warmup stage.
-        Currently it supports 'linear' and 'constant'.
     """
-    def __init__(self, mode, baselr, niters, nepochs,
-                 step=(30, 60, 90), step_factor=0.1, targetlr=0, power=0.9,
-                 warmup_epochs=0, warmup_lr=0, warmup_mode='linear'):
+    def __init__(self, mode, baselr, targetlr=0, niters=0, offset=0, power=2,
+                 step=None, step_factor=0.1):
         super(LRScheduler, self).__init__()
-        assert(mode in ['step', 'poly', 'cosine'])
-        assert(warmup_mode in ['linear', 'constant'])
+        assert(mode in ['constant', 'step', 'linear', 'poly', 'cosine'])
 
         self.mode = mode
         self.baselr = baselr
```

Comment:
the parent class uses `self.base_lr` instead of `self.baselr`
https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/lr_scheduler.py#L43

Reply:
Will match the parent class and warn users for using `self.baselr`.